### ISSUE TITLE
Can't tell apart untyped from typed class variable assignment

### ISSUE THREAD

**Issue by euresti** [python/mypy#4387]
While working on #2088 I came across an interesting problem. Take this code

```
class A:
   x: int
   y: ClassVar[int] = 67
   z = 67
```

It's clear that `x` is an instance variable and `y` is a class variable.  However `z` is a bit vague.  Technically, as far as python is concerned it's a class variable.  Both dataclasses and `attr.s(auto_attribs=True)` will ignore `z` when creating the `__init__`.  (Because they only look in `cls.__annotations__`)  (Full disclosure, I haven't run dataclasses yet but looking at the code it looks that way)  However I can't figure out a way to tell the difference in mypy code.

Basically I'm walking the class body looking for `AssignmentStmt`s.  And I thought  "Oh hey. node.is_classvar" should help me here.  It return False for `x`, True for `y`, but sadly it also returns False for `z`. To make it harder, if you look at `visit_assignment_stmt` it does `analyze_simple_literal_type` to infer the type of `z` and then `store_declared_types` sets `is_inferred_def = False` so I can't tell apart `z = 67` from `z: int = 67`

**Comment by ilevkivskyi** [comment]
I think PEP 526 is clear on this, everything that is not typed with ``ClassVar`` is an instance variable. In particular:
```python
class C:
    z = 42
```
introduces an instance variable with a default value. As for how to check this in mypy, there are several options and the simplest one is probably to re-use ``stmt.new_syntax``, see ``semanal.check_namedtuple_classdef``.

**Comment by euresti** [comment]
Ooh. `new_syntax` is exactly what I need.  Thanks!
